app/src/main/java/com/github/se/studybuddies/database/DatabaseConnection.kt

```diff
@@ -1112,6 +1125,101 @@ class DatabaseConnection {
     return TopicList(emptyList())
   }
 
+  suspend fun getAllContacts(uid: String): ContactList {
+    try {
+      val snapshot = userContactsCollection.document(uid).get().await()
+      val items = mutableListOf<Contact>()
+
+      if (snapshot.exists()) {
+        val contactsUIDs = snapshot.data?.get("contacts") as? List<String>
+        contactsUIDs?.let { contactsIDs ->
+          contactsIDs.forEach { contactID ->
+            val document = contactDataCollection.document(contactID).get().await()
+            val members = document.get("members") as? Pair<String, String> ?: Pair("", "")
+            val showOnMap = document.get("showOnMap") as Boolean
+            items.add(Contact(contactID, members, showOnMap))
+          }
+        }
+        return ContactList(items)
+      } else {
+        Log.d("MyPrint", "User with uid $uid does not exist")
+        return ContactList(emptyList())
+      }
+    } catch (e: Exception) {
+      Log.d("MyPrint", "In ViewModel, could not fetch groups with error: $e")
+    }
+    return ContactList(emptyList())
+  }
+
+  suspend fun getContact(contactUID: String): Contact {
+    val document = contactDataCollection.document(contactUID).get().await()
+    return if (document.exists()) {
+      val members = document.get("members") as? Pair<String, String> ?: Pair("", "")
+      val showOnMap = document.get("showOnMap") as Boolean
+      Contact(contactUID, members, showOnMap)
+    } else {
+      Log.d("MyPrint", "contact document not found for contact id $contactUID")
+      Contact.empty()
+    }
+  }
+
+  suspend fun createContact(otherUID: String) {
```

Comment:
I think you can create two time the same contact is there a way to ensure that only one contact of the same person is created ?

Reply:
this bug has now been solved